### PR TITLE
Update to SDK 0.17.0

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java [hunk 1]
```diff
@@ -287,17 +287,16 @@ public void addThrowable(Span span, Throwable throwable) {
...
    */
   @Deprecated
-  public static <C> Context extractWithGlobalPropagators(
-      C carrier, TextMapPropagator.Getter<C> getter) {
+  public static <C> Context extractWithGlobalPropagators(C carrier, TextMapGetter<C> getter) {
     return extract(GlobalOpenTelemetry.getPropagators(), carrier, getter);
   }
 
-  public <C> Context extract(C carrier, TextMapPropagator.Getter<C> getter) {
+  public <C> Context extract(C carrier, TextMapGetter<C> getter) {
```
Comment (trask) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/BaseTracer.java:294]: this is a nice improvement 👍

File: instrumentation/aws-lambda-1.0/library/src/test/java/io/opentelemetry/instrumentation/awslambda/v1_0/ApiGatewayProxyRequestTest.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ void resetOpenTelemetry() {
   public void shouldCreateNoopRequestIfNoPropagatorsSet() throws IOException {
     // given
     InputStream mock = mock(InputStream.class);
-    GlobalOpenTelemetry.set(OpenTelemetry.getDefault());
+    GlobalOpenTelemetry.set(OpenTelemetry.noop());
```
Comment (trask) [inline instrumentation/aws-lambda-1.0/library/src/test/java/io/opentelemetry/instrumentation/awslambda/v1_0/ApiGatewayProxyRequestTest.java:38]: 😁

File: instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/baggage/BaggageBridging.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
 public final class BaggageBridging {
 
   public static Baggage toApplication(io.opentelemetry.api.baggage.Baggage agentBaggage) {
-    BaggageBuilder applicationBaggageBuilder = Baggage.builder().setNoParent();
+    BaggageBuilder applicationBaggageBuilder = Baggage.builder();
```
Comment (trask) [inline instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/baggage/BaggageBridging.java:15]: this is better too 👍

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java [hunk 4]
```diff
@@ -63,30 +63,5 @@ private static void copySystemProperties() {
...
-      System.setProperty("otel.traces.exporter", "otlp");
-    } else if (traceExporter.equals("none")) {
-      System.clearProperty("otel.traces.exporter");
-    }
-    String metricExporter = environmentPropertiesWithCopies.getProperty("otel.metrics.exporter");
-    if (metricExporter == null) {
-      System.setProperty("otel.metrics.exporter", "otlp");
-    } else if (metricExporter.equals("none")) {
-      System.clearProperty("otel.metrics.exporter");
-    }
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java:90]: 🎉

**Review by trask — APPROVED** [review]
Awesome 🥳

File: instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/baggage/BaggageBridging.java (hunk 3 above)
Comment (anuraaga) [reply instrumentation/opentelemetry-api-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/baggage/BaggageBridging.java:15]: Yeah I was quite happy with how baggage turned out here and in the SDK's unit tests too, much simpler API.

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java (hunk 4 above)
Comment (iNikem) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java:90]: I don't understand this :) Why is this not needed anymore?

Comment (anuraaga) [reply javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/OpenTelemetryInstaller.java:90]: The SDK autoconfigure now follows the same defaults as the spec, which are the same defaults our agent expected.

**Pull request merged**